factory/alchemy.py

```diff
@@ -140,8 +140,21 @@ class Meta:
     @classmethod
     async def _create_model_async(cls, model_class, *args, **kwargs):
         session = cls._meta.sqlalchemy_session
+
+        if session is None:
+            raise RuntimeError("No session provided.")
+
         async with session.begin():
-            model = model_class(**kwargs)
-            session.add(model)
+            return await cls._save(model_class, session, args, kwargs)
+
+    @classmethod
+    async def _save(cls, model_class, session, args, kwargs):
+        session_persistence = cls._meta.sqlalchemy_session_persistence
 
-            return model
+        obj = model_class(*args, **kwargs)
+        session.add(obj)
+        if session_persistence == SESSION_PERSISTENCE_FLUSH:
+            await session.flush()
+        elif session_persistence == SESSION_PERSISTENCE_COMMIT:
+            await session.commit()
+        return obj
```

Comment:
Code was not actually saving the object to the db. 😬 

TODO:
- add a test that would have shown that
- handle `get_or_create` in async create as well, or can we ship without? (I won't have bandwidth for it)